File: 0009-palindrome-number/0009-palindrome-number.cs

public class Solution {
    public bool IsPalindrome(int x) {
        if (x < 0) {
            return false; 
        }
        int rev=x;
        int sum = 0;
        
        while (x != 0) {
            int r = x % 10;
            sum = sum * 10 + r;
            x = x / 10;
        }
        
        return rev == sum;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        else:
            original = x
            reversed_num = 0
            while x != 0:
                digit = x % 10
                reversed_num = reversed_num * 10 + digit
                x = x // 10
            return reversed_num == original